frontendProviderServices/src/main/java/com/virtuslab/gitmachete/frontend/ui/providerservice/SelectedVcsRepositoryProvider.java

```diff
@@ -0,0 +1,37 @@
+package com.virtuslab.gitmachete.frontend.ui.providerservice;
+
+import javax.swing.JComponent;
+
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.gitmachete.frontend.ui.api.repositoryselection.IGitRepositorySelectionComponentFactory;
+import com.virtuslab.gitmachete.frontend.ui.api.repositoryselection.IGitRepositorySelectionProvider;
+import com.virtuslab.gitmachete.frontend.ui.api.repositoryselection.ISelectionComponent;
+
+@Service
+public final class SelectedVcsRepositoryProvider {
```

Comment:
Not sure if a good idea... but maybe `SelectedVcsRepositoryProvider` could implement `ISelectionComponent` or `IGitRepositorySelectionProvider` via delegating to `selectionComponent`? will that simplify things a bit?